src/utils/components/DiskModal/DiskModal.tsx

```diff
@@ -1,128 +1,40 @@
-  editDraftDisk,
-} from './utils/helpers';
-import { DiskFormState, DiskModalProps } from './utils/types';
-
-const DiskModal: FC<DiskModalProps> = ({
-  createOwnerReference = true,
-  customize = false,
-  headerText,
-  initialFormData = null,
-  isEditDisk = false,
-  isEditingCreatedDisk = false,
-  isOpen,
-  onClose,
-  onSubmit,
-  onUploadedDataVolume,
-  vm,
-}) => {
-  const { t } = useKubevirtTranslation();
-  const { upload, uploadData } = useCDIUpload();
-  const isVMRunning = isRunning(vm);
-
-  const methods = useForm<DiskFormState>({
-    defaultValues: initialFormData ?? {
-      diskName: generatePrettyName('disk'),
-      ...getInitialStateDiskForm(isVMRunning),
-    },
-    mode: 'all',
-  });
-
-  const {
-    formState: { isSubmitting, isValid },
-    handleSubmit,
-  } = methods;
-
-  const diskSource = initialFormData?.diskSource;
+import { getName } from '@kubevirt-utils/resources/shared';
+import { getDataVolumeTemplates, getVolumes } from '@kubevirt-utils/resources/vm';
+
+import { getSourceFromVolume } from './utils/helpers';
+import { SourceTypes, V1DiskModalProps } from './utils/types';
+import BlankDiskModal from './BlankDiskModal';
+import ClonePVCDiskModal from './ClonePVCDiskModal';
+import ContainerDiskModal from './ContainerDiskModal';
+import HTTPDiskModal from './HTTPDiskModal';
+import OtherDiskModal from './OtherDiskModal';
+import PVCDiskModal from './PVCDiskModal';
+import RegistryDiskModal from './RegistryDiskModal';
+import UploadDiskModal from './UploadDiskModal';
+
+const DiskModalBySource = {
+  [SourceTypes.BLANK]: BlankDiskModal,
+  [SourceTypes.CLONE_PVC]: ClonePVCDiskModal,
+  [SourceTypes.EPHEMERAL]: ContainerDiskModal,
+  [SourceTypes.HTTP]: HTTPDiskModal,
+  [SourceTypes.PVC]: PVCDiskModal,
+  [SourceTypes.REGISTRY]: RegistryDiskModal,
+  [SourceTypes.UPLOAD]: UploadDiskModal,
+};
 
-  return (
-    <FormProvider {...methods}>
-      <TabModal
-        onClose={() => {
-          if (upload?.uploadStatus === UPLOAD_STATUS.UPLOADING) {
-            upload.cancelUpload().catch(kubevirtConsole.error);
-          }
-          onClose();
-        }}
-        onSubmit={() =>
-          handleSubmit((data) => {
-            if (isEditingCreatedDisk || customize)
-              return editDraftDisk(vm, initialFormData, data, onSubmit);
+const DiskModal: FC<V1DiskModalProps> = (props) => {
+  const volume = getVolumes(props.vm)?.find((v) => v.name === props.editDiskName);
+  const dataVolumeTemplate = getDataVolumeTemplates(props.vm)?.find(
+    (dv) => getName(dv) === volume?.dataVolume?.name,
+  );
 
-            if (isEditDisk)
-              return editCreatedDisk(initialFormData, data, uploadData, {
-                createOwnerReference,
-                onSubmit,
-                onUploadedDataVolume,
-                vm,
-              });
+  const Modal =
+    DiskModalBySource[props.createDiskSource || getSourceFromVolume(volume, dataVolumeTemplate)] ||
+    OtherDiskModal;
```

Comment:
can avoid it by using proxy

Reply:
:-( proxy for something so simple ?

Reply:
well proxy is just an obj and a func , im sure u will manage :)